com.unity.netcode.gameobjects/Runtime/Core/NetworkManager.cs

```diff
@@ -260,28 +262,21 @@ private void OnValidate()
                 }
             }
 
-            var activeScene = UnityEngine.SceneManagement.SceneManager.GetActiveScene();
-            var activeSceneName = activeScene.name;
-            if (!NetworkConfig.RegisteredScenes.Contains(activeSceneName))
+            if (NetworkConfig.EnableSceneManagement)
             {
-                if (NetworkLog.CurrentLogLevel <= LogLevel.Normal)
+                foreach(var sceneAsset in NetworkConfig.RegisteredSceneAssets)
                 {
-                    NetworkLog.LogWarning("Active scene is not registered as a network scene. Netcode has added it");
-                }
-
-                NetworkConfig.RegisteredScenes.Add(activeSceneName);
-                UnityEditor.EditorApplication.delayCall += () =>
-                {
-                    if (!UnityEditor.EditorApplication.isPlaying)
+                    if(!NetworkConfig.RegisteredScenes.Contains(sceneAsset.name))
```

Comment:
What if RegisteredScenes was a Set, not a List?  Or is order important?

Reply:
The order isn't important, but I am going to defer my answer to the final PR for scene management improvements:
[RFC-PR-32](https://github.com/Unity-Technologies/com.unity.multiplayer.rfcs/pull/32)
[PR-967](https://github.com/Unity-Technologies/com.unity.multiplayer.mlapi/pull/967)

_This portion will be quickly replaced (hopefully within a week and change depending upon how reviews go) with the above, which makes registered scenes obsolete._

Reply:
so, I assume you were implying that we'd expect network scene registration to go away entirely.

Reply:
Yeah.  Sorry for the ambiguity.  PR-967 makes all things "Scene Registration" relative no longer relative as all of that code will be removed.